File: arrays/jumpGame.java
package arrays;

public class jumpGame {
    public static void main(String[] args) {
        SolutionJG solution = new SolutionJG();
        System.out.println(solution.canJump(new int[]{2,3,1,1,4}));
        System.out.println(solution.canJump(new int[]{3,2,1,0,4}));
        
    }
}


class SolutionJG {
    public boolean canJump(int[] nums) {
        
        int reachable =0;
        for (int i=0 ; i< nums.length; i++){
            if (i> reachable) break;
            reachable = Math.max(reachable, i + nums[i]);
            //System.out.println(reachable);
        }
        return reachable>=nums.length-1;        
    }
}

File: dynamicProgramming/jumpGame.py
# https://leetcode.com/problems/jump-game/
# https://www.youtube.com/watch?v=muDPTDrpS28

from typing import List


class Solution:
    # Runtime: 1211 ms, faster than 13.81% of Python3 online submissions for Jump Game.
    # Memory Usage: 15.2 MB, less than 82.18% of Python3 online submissions for Jump Game.
    def canJump(self, nums: List[int]) -> bool:
    
        reachable = nums[0]
        
        for index , jump in enumerate(nums):
            if index>reachable : break
            reachable = max(index+jump, reachable)
        
        return reachable >= len(nums)-1